### DISCUSSION TITLE
Migrating to new cable model of 3.3

### DISCUSSION THREAD

**Discussion by bluikko** [netbox-community/netbox#10072] [Q&A]
I am sure I am not the only with plenty of fiber-optic duplex cables documented pre-3.3 that would now need to be migrated to the new model in 3.3.

Currently duplex fibers are modeled as follows:
- Each duplex cable is a single cable with each duplex connector modeled as a single connector.
- Duplex port ODFs and panels are modeled as a single front/back port with port name using a name convention to list both 2 core numbers.

How are other people migrating this to use 2 individual strands of fiber with real duplex ports?  
Obviously manually deleting & recreating everything is a massive undertaking. So some kind of scripting is likely involved.

Thank you for finally implementing this feature. I hope there could be some discussion and ideas on how to best migrate to use it.

**Comment by jeremystretch** [comment]
I suspect most people will employ [custom scripts](https://docs.netbox.dev/en/stable/customization/custom-scripts/) particular to their own environments.

**Reply by bluikko** [reply]
That seems like a feasible way.  
It would be interesting to see what kind of scripts other people utilize so that a generic base/example script could be improved on by several users - in a reduced amount of time, instead of everyone using an amount of time to develop their own.
